src/ios/CDVFileTransfer.m

```diff
@@ -440,13 +440,23 @@ - (void)download:(CDVInvokedUrlCommand*)command
         targetURL = [[self.commandDelegate getCommandInstance:@"File"] fileSystemURLforLocalPath:target].url;
     } else {
         targetURL = [NSURL URLWithString:target];
+
+        if (targetURL == nil) {
+            NSString* targetUrlTextEscaped = [target stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
```

Comment:
stringByAddingPercentEscapesUsingEncoding is deprecated, would be better to use [target stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLFragmentAllowedCharacterSet]]